src/services/catalog.service.ts

```diff
@@ -2,16 +2,75 @@ import * as angular from 'angular';
 import * as _ from 'lodash';
 
 export class CatalogService {
-  static $inject = ['Constants', '$filter'];
+  static $inject = ['$filter', '$q', 'Constants', 'DataService', 'Logger'];
 
   public $filter: any;
-  private constants: any;
+  public catalogServiceItems: any;
+
+  private $q: any;
   private categories: any;
+  private constants: any;
+  private dataService: any;
+  private logger: any;
 
-  constructor(constants: any, $filter: any) {
-    this.constants = constants;
+  constructor($filter: any, $q: any, constants: any, DataService: any, Logger: any) {
     this.$filter = $filter;
+    this.$q = $q;
+    this.constants = constants;
     this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+    this.dataService = DataService;
+    this.logger = Logger;
+  }
+
+  public getCatalogServiceItems() {
+    let deferred: any = this.$q.defer();
+    let serviceClasses: any, imageStreams: any;
+
+    this.getServiceClasses().then((resources: any) => {
+      serviceClasses = resources;
+      this.getImageStreams().then((resources: any) => {
```

Comment:
It seems wrong to me that if I call a method that is supposed to get me both, and it only gives me some, that it returns success. As a caller, I would want to know if something failed.

If we really care about continuing after errors, we should have separate methods for service classes and image streams and then another method to combine and sort them. Then the callers can decide what to do if one fails. I kind of like the current approach though because it will let us turn on and off templates in the catalog in one place without having to change the calling code.